clang/test/CIR/CodeGen/vectype.cpp

```diff
@@ -0,0 +1,15 @@
+// RUN: %clang_cc1 -std=c++17 -triple x86_64-unknown-linux-gnu -fclangir-enable -emit-cir %s -o - | FileCheck %s
+
+typedef int int4 __attribute__((vector_size(16)));
+int main() {
+  int4 a = { 1, 2, 3, 4 };
+  int4 b = { 5, 6, 7, 8 };
+  int4 c = a + b;
+  return c[1];
+}
+
```

Comment:
I had started down the path of implementing a `#cir.vec` attribute for constant vector objects.  But I struggled with it, especially with figuring out when and where to create a `cir.const(#cir.vec<...>)`.  Since the vector op that I did add works to create both const and non-const vector objects, I dropped the attribute for now.

I am planning to implement `#cir.vec` in a later change.